syftlib/src/main/java/org/openmined/syft/Syft.kt

```diff
@@ -8,19 +8,44 @@ import org.openmined.syft.network.NetworkMessage
 import org.openmined.syft.network.SignallingClient
 import org.openmined.syft.threading.ProcessSchedulers
 
-class Syft(
+class Syft private constructor(
     private val workerId: String,
     private val keepAliveTimeout: Int = 20000,
     private val url: String,
     private val schedulers: ProcessSchedulers
 ) {
-
     private lateinit var signallingClient: SignallingClient
     private val compositeDisposable = CompositeDisposable()
 
+    companion object {
+        operator fun invoke(
+            workerId: String,
+            keepAliveTimeout: Int = 20000,
+            url: String,
+            schedulers: ProcessSchedulers
+        ): Syft? =
+                Syft(workerId, keepAliveTimeout, url, schedulers).takeIf { isValid(it) }
+
+        private fun isValid(syft: Syft): Boolean {
+            return checkProtocol(syft.url) &&
+                   checkKeepAliveTimeout(syft.keepAliveTimeout)
+        }
+
+        private fun checkProtocol(url: String) =
+                if (!url.startsWith("wss", true)) {
+                    println("Protocol must be wss")
+                    false
+                } else true
+
+        private fun checkKeepAliveTimeout(timeout: Int) =
```

Comment:
Eventually, I wish to use unsigned integer here when it gets out of experimental for the same reasons as above

Reply:
We can change it when we need it later.